src/ClientContext.ts

```diff
@@ -138,6 +141,45 @@ export class ClientContext {
     return this.processQueryFeedResponse(response, !!query, resultFn);
   }
 
+  public async getQueryPlan(
+    path: string,
+    resourceType: ResourceType,
+    resourceId: string,
+    query: SqlQuerySpec | string,
+    options: FeedOptions = {}
+  ): Promise<Response<PartitionedQueryExecutionInfo>> {
+    const request: RequestContext = {
+      globalEndpointManager: this.globalEndpointManager,
+      requestAgent: this.cosmosClientOptions.agent,
+      connectionPolicy: this.connectionPolicy,
+      method: HTTPMethod.post,
+      path,
+      operationType: OperationType.Read,
+      client: this,
+      resourceId,
+      resourceType,
+      options,
+      body: query,
+      plugins: this.cosmosClientOptions.plugins
+    };
+
+    request.endpoint = await this.globalEndpointManager.resolveServiceEndpoint(request);
+    request.headers = await this.buildHeaders(request);
+    request.headers[Constants.HttpHeaders.IsQueryPlan] = "True";
+    request.headers[Constants.HttpHeaders.QueryVersion] = "1.4";
+    request.headers[Constants.HttpHeaders.SupportedQueryFeatures] =
+      "Aggregate, Distinct, MultipleOrderBy, OffsetAndLimit, OrderBy, Top, CompositeAggregate";
+    request.headers[Constants.HttpHeaders.ContentType] = Constants.MediaTypes.QueryJson;
+    if (typeof query === "string") {
+      request.body = { query }; // Converts query text to query object.
```

Comment:
I think this should be in the format:

```
{
    "query" : <query>,
    "parameters": []
}
```

Can you add one test to see that the query plan code work with parameters?

Reply:
We test this in some of the cross partition specs already